roles/ceph-container-common/tasks/prerequisites.yml

```diff
@@ -1,4 +1,22 @@
 ---
+- name: Install selinux modules
+  when:
+    - ansible_facts['distribution'] == 'RedHat'
+    - inventory_hostname in groups.get(mon_group_name, [])
+      or inventory_hostname in groups.get(rgw_group_name, [])
+  block:
+    - import_role:
+        name: ceph-common
+        tasks_from: configure_repository.yml
+
+    - name: install ceph-selinux package
+      package:
+        name: ceph-selinux
+        state: present
+
+    - name: restore certificates selinux context
+      command: /usr/sbin/restorecon -RF /etc/pki/ca-trust/extracted 
```

Comment:
Would you please say more about the problem?

What file in `/etc/pki/ca-trust/extracted` is mis-labeled?

What is the thing that's mis-labeling it in the first place?

Reply:
As the mon and rgw containers need to get access to the local folder `/etc/pki/ca-trust/extracted` and its content, this line [https://github.com/ceph/ceph-ansible/blob/c491e67486d8ed2717b5b9eda96544eb4f7eae2d/roles/ceph-rgw/templates/ceph-radosgw.service.j2#L42](url) relabels the whole folder content with container_t at container startup. While it gives access to the container(s) it prevents local processes like sssd to access the files. Unfortunately just changing the flag from z to ro does not put back the cert_t label on the files, this is why that restorecon is necessary.

I hope having answered your question.

Reply:
see https://github.com/ceph/ceph/pull/44802#issuecomment-1024234838